app/controllers/Checkout.scala

```diff
@@ -45,6 +45,7 @@ object Checkout extends Controller {
           yield BadRequest(views.html.checkout.payment(formWithErrors, userIsSignedIn = idUserOpt.isDefined))
       },
       userData => {
+        CheckoutService.processSubscription(userData, AuthenticationService.authenticatedUserFor(request))
```

Comment:
It looks like if `processSubscription` fails, the user is still taken to the "thank you" page. Please map this to the redirect, potentially with a fallback on rendering the checkout form in case of error.